media/server/gstplayer/source/tasks/generic/HandleBusMessage.cpp

```diff
@@ -196,6 +196,72 @@ void HandleBusMessage::execute() const
         m_glibWrapper->gErrorFree(err);
         break;
     }
+    case GST_MESSAGE_WARNING:
+    {
+        PlaybackError rialtoError = PlaybackError::UNKNOWN;
+        GError *err = nullptr;
+        gchar *debug = nullptr;
+        m_gstWrapper->gstMessageParseWarning(m_message, &err, &debug);
+
+        if ((err->domain == GST_STREAM_ERROR) && (err->code == GST_STREAM_ERROR_DECRYPT))
+        {
+            RIALTO_SERVER_LOG_WARN("Decrypt error %s - %d: %s (%s)", GST_OBJECT_NAME(GST_MESSAGE_SRC(m_message)),
+                                   err->code, err->message, debug);
+            rialtoError = PlaybackError::DECRYPTION;
+        }
+        else
+        {
+            RIALTO_SERVER_LOG_WARN("Unknown warning, ignoring %s - %d: %s (%s)",
+                                   GST_OBJECT_NAME(GST_MESSAGE_SRC(m_message)), err->code, err->message, debug);
+        }
+
+        if ((PlaybackError::UNKNOWN != rialtoError) && (m_gstPlayerClient))
+        {
+            const gchar *mimetype = nullptr;
```

Comment:
What do you think about giving names to decryptors from which you would be able to get the media type?
For example audio_decryptor (you might also add a number in similar why like we do with appsrc elements)
This means you would have to pass a non-null name in _createDecryptorElement_.
I think that would be less complicated and less error prone code.

Reply:
Yeah that makes sense, I think deliver this and ill do that in a followup.